ion-schema/src/main/kotlin/com/amazon/ionschema/internal/TypeReference.kt

```diff
@@ -80,22 +84,23 @@ internal class TypeReference private constructor() {
                 is IonSymbol -> handleSymbol(ion, schema, referenceManager)
                 else -> throw InvalidSchemaException("Unable to resolve type reference '$ion'")
             }
+
+            val maybeNullableType = handleNullable(ion, schema, theType)
```

Comment:
🗺️ There's a lot of branching in this class to handle different sorts of type references. I refactored the class so that `handleNullable()` only needs to be called in one place (so that it doesn't get missed for any branches).

Basically, this means that all other calls to `handleNullable()` are removed, and all of the private functions now return `TypeInternal` instead of `() -> TypeInternal`.